EREGCSC-656 In search, left sidebar has list of parts

cgodwin1 opened the pull request:
* Added get_structure func to utils
   - Merges structures to acquire full list of parts
   - Moved from homepage.py
   - Should core do this instead?
* Added 'left_sidebar_content' parameter to context
   - Problem: Default left sidebar view not correct for full list of parts
   - Solution: Allow different templates for different data types
   - Downside: All views utilizing the left sidebar must specify a template :(

Commit:
Draft of left sidebar in search

* Added get_structure func to utils
   - Merges structures to acquire full list of parts
   - Moved from homepage.py
   - Should core do this instead?
* Added 'left_sidebar_template' parameter to context
   - Problem: Default left sidebar view not correct for search
   - Solution: Allow different templates for different data types
   - Downside: All views must specify a template :(

Commit:
Added styling for search left sidebar

Commit:
Added links to parts

whytheplatypus commented on regulations/views/reader.py:
```diff
@@ -39,6 +39,7 @@ def get_context_data(self, **kwargs):
             'toc':          toc,
             'parts':        parts,
             'versions':     versions,
+            'left_sidebar_content': 'regulations/partials/left_sidebar_views/subpart_view.html',
```
A custom left sidebar for the search view will be more clear and allow all of this logic to remain at the template level.

cgodwin1 replied:
Implemented in d695d5a

whytheplatypus commented:
heads up that the merge of https://github.com/CMSgov/regulations-site/pull/70 will probably require updates to the url generation in this PR

Commit:
Use template inheritance for different left sidebar views

Variations of the left sidebar now inherit from `regulations/partials/sidebar_left/base.html`,
and override block `navigation`.

This results in cleaner code than the previous solution, and keeps template logic
at the template level.

Commit:
Merge remote-tracking branch 'upstream/main' into search-list-parts

whytheplatypus commented on regulations/templates/regulations/partials/sidebar_left/base.html:
```diff
@@ -0,0 +1,40 @@
+
+        <form method="GET" action="{% url 'goto' %}">
+        <input name="{{ reg_part }}-version" type="hidden" required value={{ version }}>
+        <input name="title" type="hidden" required value={{ title }}>
```
This is coming up blank in this case breaking to the Jump To behavior. We may have to rethink how this works here since Search doesn't have a specific title to fall back on.

cgodwin1 replied:
jump-to removed from search sidebar

Commit:
Merge remote-tracking branch 'upstream/main' into search-list-parts

Commit:
Remove jump-to from search sidebar
